linera-execution/src/system.rs

```diff
@@ -556,11 +561,14 @@ where
             PublishDataBlob { blob_hash } => {
                 self.blob_published(&BlobId::new(blob_hash, BlobType::Data))?;
             }
-            PublishCommitteeBlob { blob_hash } => {
-                self.blob_published(&BlobId::new(blob_hash, BlobType::Committee))?;
-            }
             ReadBlob { blob_id } => {
-                self.read_blob_content(blob_id).await?;
+                let content = self.read_blob_content(blob_id).await?;
+                if blob_id.blob_type == BlobType::Data {
```

Comment:
Why do we not track the other two `BlobType`s as well?

Reply:
* We want all other chains to keep with the latest epoch as much as possible, and the only way to read a `Committee` blob is to do exactly that. Each chain can only read each `Committee` blob exactly once (or: as often as it occurs in different epochs), and we want them all to do that.
* For application description and bytecode blobs it's a bit debatable, but it's not a great user experience if the price of executing a transaction or operation, before the execution even starts, already depends on the size of the bytecode or parameters. Also, we cache modules in the chain worker, I think, so the reading cost for the validator only occurs whenever they had evicted that chain worker and are reloading it, which isn't deterministic.